src/hable.js

```diff
@@ -53,6 +67,20 @@ export default class Hookable {
     for (const key in hooks) {
       this.hook(key, hooks[key])
```

Comment:
We can use a map fn here to have an array of callbacks and return a function to all all of them + unref. This way can avoid duplicate logic.